ext/flash/01_http.js

```diff
@@ -227,6 +228,192 @@
+      const responseStr = http1Response(
+        method,
+        innerResp.status ?? 200,
+        innerResp.headerList,
+        respBody,
+        length,
+      );
+      writeFixedResponse(
+        serverId,
+        i,
+        responseStr,
+        length,
+        !ws, // Don't close socket if there is a deferred websocket upgrade.
+        respondFast,
+      );
+    }
+
+    (async () => {
+      if (!ws) {
+        if (hasBody && body[_state] !== "closed") {
+          // TODO(@littledivy): Optimize by draining in a single op.
+          try {
+            await req.arrayBuffer();
+          } catch { /* pass */ }
+        }
+      }
+
+      if (isStreamingResponseBody === true) {
+        const resourceRid = getReadableStreamRid(respBody);
+        if (resourceRid) {
+          if (respBody.locked) {
+            throw new TypeError("ReadableStream is locked.");
+          }
+          const reader = respBody.getReader(); // Aquire JS lock.
+          try {
+            core.opAsync(
+              "op_flash_write_resource",
+              http1Response(
+                method,
+                innerResp.status ?? 200,
+                innerResp.headerList,
+                0, // Content-Length will be set by the op.
+                null,
+                true,
+              ),
+              serverId,
+              i,
+              resourceRid,
+            ).then(() => {
+              // Release JS lock.
+              readableStreamClose(respBody);
+            });
+          } catch (error) {
+            await reader.cancel(error);
+            throw error;
+          }
+        } else {
+          const reader = respBody.getReader();
+          writeFixedResponse(
+            serverId,
+            i,
+            http1Response(
+              method,
+              innerResp.status ?? 200,
+              innerResp.headerList,
+              respBody.byteLength,
+              null,
+            ),
+            respBody.byteLength,
+            false,
+            respondFast,
+          );
+          while (true) {
+            const { value, done } = await reader.read();
+            await respondChunked(
+              i,
+              value,
+              done,
+            );
+            if (done) break;
+          }
+        }
+      }
+
+      if (ws) {
+        const wsRid = await core.opAsync(
+          "op_flash_upgrade_websocket",
+          serverId,
+          i,
+        );
+        ws[_rid] = wsRid;
+        ws[_protocol] = resp.headers.get("sec-websocket-protocol");
+
+        ws[_readyState] = WebSocket.OPEN;
+        const event = new Event("open");
+        ws.dispatchEvent(event);
+
+        ws[_eventLoop]();
+        if (ws[_idleTimeoutDuration]) {
+          ws.addEventListener(
+            "close",
+            () => clearTimeout(ws[_idleTimeoutTimeout]),
+          );
+        }
+        ws[_serverHandleIdleTimeout]();
+      }
+    })().catch(console.error);
```

Comment:
Question:
In the original this async function would call the onError handler. It doesn't appear that the responseHandler function awaits or returns this whole async block so the `.catch(onError)` on line 557 doesn't handle this case.

Is this expected?

Reply:
Oh that is an oversight